hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/ApproveAllowanceValidator.java

```diff
@@ -181,6 +181,10 @@ private void validateNftAllowances(
             final var spenderAccount = accountStore.getAccountById(spender);
             validateNFTSpender(serialNums, spenderAccount);
 
+            if (Boolean.TRUE.equals(allowance.approvedForAll())) {
```

Comment:
Nice catch, this definitely seems right. 🙌 Was there a `@HapiTest` failing, or you just noticed this by inspection?

Reply:
Thanks! Something didn't seem right. When I gave an input with an invalid account ID, the response was always 'SUCCESS' for setApprovalForAll, but for Grant approval it was returning correctly INVALID_ALLOWANCE_SPENDER_ID. So, I inspected the flow and found out that a similar check exists in mono, but is missing here.